extension/scripts/content/city/ttCity.js

```diff
@@ -0,0 +1,189 @@
+"use strict";
+
+(async () => {
+    await loadDatabase();
+    if (!settings.pages.city.highlight && !settings.pages.city.showValue) return;
+    
+    console.log("TT - City: Starting...");
+    await requireMapLoaded();
+    run();
+    listenToSettings();
+	console.log("TT - City: Ready");
+})();
+
+function requireMapLoaded() {
+	return requireElement("#map .leaflet-marker-pane .highlightItemMarket");
+}
+
+function run() {
+    let container;
+	if (settings.pages.city.showValue) {
+		container = createItemContainer();
+	}
+
+	if (settings.pages.city.highlight) {
+		if (settings.pages.city.closedHighlight) {
+            addHighlightForItems();
+		} else if (container) {
+            if (!isContainerClosed(container)) {
+				addHighlightForItems();
+			}
+			addContainerClosedEventListenerForHighlightOfItems(container);
+		}
+	}
+}
+
+function listenToSettings() {
+	storageListeners.settings.push((_) => {
+        reset();
+        run();
+    });
+}
+
+function reset() {
+	document.find("#tt-city-items-container")?.remove();
```

Comment:
I love optional chaining.Sadly enough, Kiwi Browser sucks and doesn't support it.

Reply:
Addressed in 254aa39c6c6983c71a1077959d31c29c7215da37